testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -352,46 +426,122 @@ def test_atomgroup_to_atomgroup(self, u):
+            _ = atm._cache['unique']
+        # populate uniqueness cache of res:
+        assert res.isunique
+        atm = res.atoms
+        # assert uniqueness caches of atm are set:
+        assert atm._cache['isunique'] is True
+        assert atm._cache['unique'] is atm
 
     def test_residuegroup_to_residuegroup(self, u):
         res = u.residues.residues
         assert len(res) == 25
         assert isinstance(res, groups.ResidueGroup)
+        assert res is u.residues
 
     def test_residuegroup_to_segmentgroup(self, u):
         seg = u.residues.segments
         assert len(seg) == 5
         assert isinstance(seg, groups.SegmentGroup)
+        assert seg == u.segments
+        assert seg is not u.segments
+        assert seg._cache['isunique'] is True
+        assert seg._cache['unique'] is seg
 
     def test_segmentgroup_to_atomgroup(self, u):
-        atm = u.segments.atoms
+        seg = u.segments
+        atm = seg.atoms
         assert len(atm) == 125
         assert isinstance(atm, groups.AtomGroup)
+        assert atm == u.atoms
+        assert atm is not u.atoms
+        # clear seg's uniqueness caches:
+        if 'unique' in seg._cache.keys():
+            del seg._cache['unique']
+        if 'isunique' in seg._cache.keys():
+            del seg._cache['isunique']
+        atm = seg.atoms
+        # assert uniqueness caches of atm are empty:
+        with pytest.raises(KeyError):
+            _ = atm._cache['isunique']
+        with pytest.raises(KeyError):
+            _ = atm._cache['unique']
+        # populate uniqueness cache of seg:
+        assert seg.isunique
+        atm = seg.atoms
+        # assert uniqueness caches of atm are set:
+        assert atm._cache['isunique'] is True
+        assert atm._cache['unique'] is atm
 
     def test_segmentgroup_to_residuegroup(self, u):
-        res = u.segments.residues
+        seg = u.segments
+        res = seg.residues
         assert len(res) == 25
         assert isinstance(res, groups.ResidueGroup)
+        assert res == u.residues
+        assert res is not u.residues
+        # clear seg's uniqueness caches:
+        if 'unique' in seg._cache.keys():
+            del seg._cache['unique']
+        if 'isunique' in seg._cache.keys():
+            del seg._cache['isunique']
+        res = seg.residues
+        # assert uniqueness caches of res are empty:
+        with pytest.raises(KeyError):
+            _ = res._cache['isunique']
+        with pytest.raises(KeyError):
+            _ = res._cache['unique']
+        # populate uniqueness cache of seg:
+        assert seg.isunique
+        res = seg.residues
+        # assert uniqueness caches of res are set:
+        assert res._cache['isunique'] is True
+        assert res._cache['unique'] is res
 
     def test_segmentgroup_to_segmentgroup(self, u):
         seg = u.segments.segments
         assert len(seg) == 5
         assert isinstance(seg, groups.SegmentGroup)
+        assert seg is u.segments
```

Comment:
@orbeckst equivalent to `assert sg.segments is sg`

Reply:
ok, will change my commit